test/ATokenCheckpointed.t.sol

```diff
@@ -0,0 +1,1974 @@
+
+    // Try to submit votes on behalf of the pool. It should fail.
+    vm.expectRevert(bytes("no votes expressed"));
+    aToken.castVote(_proposalId);
+
+    // Express voting preference on the proposal.
+    vm.prank(_who);
+    aToken.expressVote(_proposalId, _supportType);
+
+    // Now votes should be castable.
+    aToken.castVote(_proposalId);
+  }
+
+  function _testCastVoteWithTransferredATokens(
+    address _userA,
+    address _userB,
+    uint256 _weight,
+    uint256 _transferAmount,
+    uint8 _supportTypeA,
+    uint8 _supportTypeB
+  ) private {
+    // Deposit some funds.
+    _mintGovAndSupplyToAave(_userA, _weight);
+    assertEq(aToken.balanceOf(_userA), _weight);
+    assertEq(aToken.balanceOf(_userB), 0);
+
+    // Transfer all aTokens from userA to userB.
+    vm.prank(_userA);
+    aToken.transfer(_userB, _transferAmount);
+    assertEq(aToken.balanceOf(_userA), _weight - _transferAmount);
+    assertEq(aToken.balanceOf(_userB), _transferAmount);
+
+    // Advance one block so that our votes will be checkpointed by the govToken;
+    vm.roll(block.number + 1);
+
+    // Create the proposal.
+    uint256 _proposalId = _createAndSubmitProposal();
+
+    // Express voting preferences.
+    if (aToken.balanceOf(_userA) == 0) vm.expectRevert(bytes("no weight"));
+    vm.prank(_userA);
+    aToken.expressVote(_proposalId, _supportTypeA);
+    vm.prank(_userB);
+    aToken.expressVote(_proposalId, _supportTypeB);
+
+    // Wait until after the pool's voting period closes.
+    vm.roll(aToken.internalVotingPeriodEnd(_proposalId) + 1);
+
+    // Submit votes on behalf of the pool.
+    aToken.castVote(_proposalId);
+
+    (uint256 _againstVotes, uint256 _forVotes, uint256 _abstainVotes) =
+      governor.proposalVotes(_proposalId);
+
+    if (_supportTypeA == _supportTypeB) {
+      if (_supportTypeA == uint8(VoteType.For)) assertEq(_forVotes, _weight);
+      if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weight);
+      if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weight);
+    } else {
+      // forgefmt: disable-start
+      if (_supportTypeA == uint8(VoteType.For)) assertEq(_forVotes, _weight - _transferAmount);
+      if (_supportTypeA == uint8(VoteType.Against)) assertEq(_againstVotes, _weight - _transferAmount);
+      if (_supportTypeA == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _weight - _transferAmount);
+      if (_supportTypeB == uint8(VoteType.For)) assertEq(_forVotes, _transferAmount);
+      if (_supportTypeB == uint8(VoteType.Against)) assertEq(_againstVotes, _transferAmount);
+      if (_supportTypeB == uint8(VoteType.Abstain)) assertEq(_abstainVotes, _transferAmount);
+      // forgefmt: disable-end
+    }
+  }
+}
+
+contract GetPastStoredBalanceTest is AaveAtokenForkTest {
+  function test_GetPastStoredBalanceCorrectlyReadsCheckpoints() public {
+    _initiateRebasing();
+
+    address _who = address(0xBEEF);
```

Comment:
similar comment about `makeAddr` instead

Reply:
Done in abaa5ff